proposals/extensions.md

```diff
@@ -353,14 +344,15 @@ Note: we may choose to only emit one extension skeleton type in metadata when du
 
 The method bodies for method/property/indexer declarations in an extension declaration in source are emitted 
 as static implementation methods in the top-level static class.  
-- An implementation method is named by prepending `<Extension>` for instance case or `<StaticExtension>` for static case to the name of the original method.  
-  For example: `set_Property` => `<Extension>set_Property`.
+- An implementation method has the same name as the original method.  
 - It has type parameters derived from the extension declaration prepended to the type parameters of the original method (including attributes).  
 - It has the same accessibility and attributes as the original method.  
 - If it implements a static method, it has the same parameters and return type. 
 - It if implements an instance method, it has a prepended parameter to the signature of the original method. 
   This parameter's attributes, refness, type, and name are derived from the receiver parameter declared in the relevant extension declaration.
 - The parameters in implementation methods refer to type parameters owned by implementation method, instead of those of an extension declaration.  
+- If the original member is an instance method, the implementation method is marked with an `[Extension]` attribute.
+- If the original member is static, the return type of the implementation method is marked with `modopt` to the parameter type of the containing extension declaration.
```

Comment:
What's the purpose of emitting a `modopt`? For metadata roundtrip we can use the `<>E__` skeleton nested types, right?

Reply:
Using a `modopt` allows us to emit both of these names as speakable. 

```csharp
static class E {
  extension (C1)
    static C1 Create(); 
  extension (C2)
    static C2 Create();
}
```

Using `modopt` on a variation of the extension type it means that we can emit what are otherwise _identical_ signatures and keep speakable names. In the case there are no conflicts other languages can invoke the methods without any additional logic. In the case there are conflicts then other compilers will need to decide how they handle it.

Reply:
I'd suggest explaining that in the speclet. But later in the speclet we seem to state that the language will not support this?

Reply:
@jaredpar In your example above, I don't think we need modopts, because the return types of the factories are different.

The case where you'd need modopt is when having the same return type. Is that right?
```
static class E {
  extension (C1)
    static Type Create(); 
  extension (C2)
    static Type Create(); // signatures conflict unless we change the signature (for example, using modopts)
}
```